avogadro/qtplugins/select/select.cpp

```diff
@@ -544,6 +550,9 @@ void Select::createLayerFromSelection()
   if (!m_molecule)
     return;
 
+  if(!m_elements)
```

Comment:
This is definitely not the right fix. That's a dialog if someone wants to select elements. If someone opens the "Select by Elements" the dialog will be created.

I'd check to see if the molecule has atoms?

Reply:
I can use m_molecule->atomCount() to check this

Reply:
but i think we need to select atoms first to create a newLayer

Reply:
thats the reason i went with this solution

Reply:
I have made relevant changes  you can check it

Reply:
I have updated according to the latest PR of issue 1499